### PR TITLE
feat(oracle): add support for loading Oracle RAW and BLOB types

### PR DESCRIPTION
**Pull request by gforsyth**
Need to add a test for the `RAW` and `BLOB` loading and I don't like the fix for the column inference fallback

### CODE REVIEW

**Comment by gforsyth** [comment]
Tests added and I'm much happier with this fix than the previous version.

File: ibis/backends/oracle/__init__.py [hunk 1]
```diff
@@ -179,6 +180,18 @@ def _metadata(self, query: str) -> Iterable[tuple[str, dt.DataType]]:
 
         name = util.gen_name("oracle_metadata")
 
+        # If query is a table, adjust the query accordingly
+        sg_expr = sg.parse_one(query, dialect="oracle")
```
Comment (cpcloud) [inline ibis/backends/oracle/__init__.py:184]: To make this maybe a bit less confusing, you can tell sqlglot how to parse the query:

```python
try:
    sg_expr = sg.parse_one(query, into=sg.exp.Table, dialect="oracle")
except ParseError:  # or whatever this error should be
    sg_expr = sg.parse_one(query, dialect="oracle")
```

It's a bit repetitive, but I think the intent is a bit more clear ("try to parse this query as a table first")

Comment (cpcloud) [reply ibis/backends/oracle/__init__.py:184]: That said, this isn't a blocker!

**Review by cpcloud — APPROVED** [review]

File: ibis/backends/oracle/__init__.py (hunk 1 above)
Comment (gforsyth) [reply ibis/backends/oracle/__init__.py]: This is a good suggestion and requires less head-scratching.  Updated!

**Pull request merged**